protocol/segment.go

```diff
@@ -30,49 +31,61 @@ type segment struct {
 	timestamp      time.Time
 }
 
-func (seg *segment) dataOffset() byte {
-	return seg.buffer[indexDataOffset]
+func (seg *segment) getDataOffset() byte {
+	return seg.buffer[dataOffsetPosition.Start]
 }
 
-func (seg *segment) headerSize() int {
-	return int(seg.dataOffset())
+func (seg *segment) getHeaderSize() int {
+	return int(seg.getDataOffset())
 }
 
-func (seg *segment) flags() byte {
-	return seg.buffer[indexFlags]
+func (seg *segment) getFlags() byte {
+	return seg.buffer[flagPosition.Start]
 }
 
-func (seg *segment) flaggedAs(flag byte) bool {
-	return seg.flags()&flag == flag
+func (seg *segment) isFlaggedAs(flag byte) bool {
+	return seg.getFlags()&flag == flag
 }
 
 func (seg *segment) getSequenceNumber() uint32 {
 	return bytesToUint32(seg.sequenceNumber)
 }
 
 func (seg *segment) getExpectedSequenceNumber() uint32 {
-	return bytesToUint32(seg.data[:seqNumberLength])
+	return bytesToUint32(seg.data[sequenceNumberPosition.Start:sequenceNumberPosition.End])
```

Comment:
Hier sollte nicht die sequence number im header zurückgegeben werden, sondern die die mit jedem ACK im payload mitgeschickt wird. Das ist immer die sequence number des nächsten pakets das erwartet wird.

Reply:
Das ist ein Fehler, der im Test [TestGetExpectedSequenceNumber(...)](https://github.com/nicosta1132/go-protocol/blob/d52347a6031b00b8515ee99bd1863997b0ce1959/protocol/segment_test.go#L46) nun sichtbar wird. 

Die Nutzdaten, wie in der ursprünglichen Version, sollten m.E. nicht direkt interpretiert werden, weil allenfalls ein ACK auch mehrere Segmente bestätigt. Dann weiss das Protokoll nicht, ob ein Datensegment oder ein ACK-Segment folgen wird.

Reply:
Was meinst du damit welche Art von Segment folgen wird? ACK Segemente werden durch ein ACK flag identifiziert, so ein Segment soll immer die nächste erwartete sequence number mitsenden. Man könnte das in einem Header field machen, das scheint mir aber eher unnötig.